entity-framework/core/miscellaneous/cli/dbcontext-creation.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Design-time DbContext Creation - EF Core
+author: bricelam
+ms.author: bricelam
+ms.date: 10/27/2017
+ms.technology: entity-framework-core
+---
+Design-time DbContext Creation
+==============================
+Some of the EF Tools commands require a DbContext instance to be created at design time. There are various ways the
+tools will try an do this.
+
+From application services
+-------------------------
+If your startup project is an ASP.NET Core app, the tools will try to obtain the DbContext object from the application's
+service provider. This is done by invoking `Program.BuildWebHost()` and accessing the `IWebHost.Services` property. Any
```

Comment:
Maybe link to ASP.NET  docs that describe the BuildWebHost pattern?

Reply:
Will add. (If I can find one.)

Reply:
There is good coverage of this in https://docs.microsoft.com/en-us/aspnet/core/migration/1x-to-2x/#update-main-method-in-programcs, but I don't think there is a stand-alone topic that describes it.